pkg/database/billing.go

```diff
@@ -0,0 +1,126 @@
+package database
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/database/cosmosdb"
+)
+
+type billing struct {
+	c    cosmosdb.BillingDocumentClient
+	uuid string
+}
+
+// Billing is the database interface for BillingDocuments
+type Billing interface {
+	Create(context.Context, *api.BillingDocument) (*api.BillingDocument, error)
+	Get(context.Context, string) (*api.BillingDocument, error)
+	Patch(context.Context, string, func(*api.BillingDocument) (bool, error)) (*api.BillingDocument, error)
+}
+
+// NewBilling returns a new Billing
+func NewBilling(ctx context.Context, uuid string, dbc cosmosdb.DatabaseClient, dbid, collid string) (Billing, error) {
+	collc := cosmosdb.NewCollectionClient(dbc, dbid)
+
+	triggers := []*cosmosdb.Trigger{
+		{
+			ID:               "setCreationBillingTimeStamp",
+			TriggerOperation: cosmosdb.TriggerOperationAll,
+			TriggerType:      cosmosdb.TriggerTypePre,
+			Body: `function trigger() {
+	var request = getContext().getRequest();
+	var body = request.getBody();
+	var date = new Date();
+	var now = Math.floor(date.getTime() / 1000);
+	var billingBody = body["billing"];
+	billingBody["creationTime"] = now;
+	billingBody["lastBillingTime"] = now;
```

Comment:
this seems dodgy
1) I'd expect this to be "if creationTime is empty, set it, otherwise don't"
2) if it's *really* wanted for us to initialise lastBillingTime, OK, but I feel like the design would be safer if we didn't touch that field in the RP at all (@shrutir25)

Reply:
Was thinking that it needs to be initiated once during creation, but i can double check with @shrutir25

Reply:
For 2) I am fine with that, the billing service can pick up the value for `lastBillingTimeStamp` from creationTime and that way RP does not have to touch `lastBillingTimeStamp` at all.